### PR TITLE
MNT: rerender

### PR DESCRIPTION
**Pull request by conda-forge-linter**
Hi! This is the friendly automated conda-forge-webservice.

I've rerendered the recipe as instructed in #17.

Here's a checklist to do before merging.
- [x] Bump the build number if needed.

### CODE REVIEW

File: .ci_support/linux_c_compilergcccxx_compilergxx.yaml
Comment (bgruening) [inline .ci_support/linux_c_compilergcccxx_compilergxx.yaml:2]: @bluescarni this is what I was hopeing for!

**Comment by bgruening** [comment]
@bluescarni would be nice if you can look at the failure.

**Comment by bluescarni** [comment]
@bgruening That pin to a specific boost version was something conda specific at one point in the past (I think it had something to do with the old way of pinning in conda-forge). piranha does not depend on any specific boost version (as long as it is recent enough). I'll try to remove the pin.

**Comment by bluescarni** [comment]
Ah wait, my bad. The pin has been removed from ``meta.yaml`` already. Need to investigate more then.

**Comment by bluescarni** [comment]
It looks like this might be a problem or bug in boost, related to whether boost has been compiled with C++14 support or not. See:

https://github.com/boostorg/system/issues/26

I pushed a tentative workaround.

Is boost compiled with ``-std=c++14`` or later in the conda-forge builds for gcc7?

**Comment by bgruening** [comment]
It should be. Here are the details: https://github.com/conda-forge/boost-cpp-feedstock/blob/master/recipe/build.sh

**Comment by bgruening** [comment]
isn't boost by default c++14 nowadays? If not we might want to add this to the boost feedstock?

**Comment by bluescarni** [comment]
I don't think so. I think it still defaults to C++98, unless you explicitly set the standard.

I don't know what would be the consequences of switching the conda-forge gcc7 boost feedstock to C++14. I am afraid there could be ABI issues popping up with the compiled boost libraries :(

**Comment by isuruf** [comment]
gcc7 builds have -std=c++17 in CXXFLAGS

**Comment by bluescarni** [comment]
@isuruf Cheers. So the problem here might be that the piranha build system sets ``-std=c++11`` when compiling the tests, and then links to a boost library compiled with ``-std=c++17``, hence the linker errors.

**Comment by bluescarni** [comment]
In the conda scripts, is there a way I can detect reliably whether the build environment is the new gcc7 one or the old gcc48 one?

**Comment by bgruening** [comment]
You can look at the logs. They show you what is installed and which compilers are being pulled.

**Comment by isuruf** [comment]
@bluescarni, you could prepend -std=c++11 or you can check CMAKE_CXX_FLAGS for -std flags

**Comment by bluescarni** [comment]
I am ok with the current workaround, which seems to be working, if you guys also are. Because piranha is header-only, the issue was affecting the compilation of some tests only.

For projects depending on piranha, I think we will have to sort out issues arising from inconsistent ``-std`` settings downstream.

**Comment by isuruf** [comment]
Ah, you meant in conda scripts.

You can check whether cxx_compiler env variable is gxx or not. If it's gxx then it's gcc7 on Linux. If it's clangxx it's clang4 on osx with -std=c++14. If it's toolchain_cxx it's the system compilers

**Comment by bluescarni** [comment]
@isuruf ok, thanks.

BTW, what is the deal with that smattering of funky C(XX)FLAGS in the gcc7 builds? E.g., I am reading this in the logs:

```console
CXXFLAGS=-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2
```

Why nocona and haswell? And why forcing ``-ftree-vectorize``?

**Comment by isuruf** [comment]
I think nocona is the oldest x86_64 processor architecture. tuning for haswell is because it's common I think.  
I've no idea why `-ftree-vectorize` is forced.

**Comment by bgruening** [comment]
@isuruf @bluescarni you are both maintainers. Your call :)

File: recipe/build.sh
Comment (isuruf) [inline recipe/build.sh]: Can you do `export CXXFLAGS="$CXXFLAGS -DBOOST_ERROR_CODE_HEADER_ONLY"` instead of this?

Comment (bluescarni) [reply recipe/build.sh]: Ok.

**Comment by bluescarni** [comment]
> I think nocona is the oldest x86_64 processor architecture. tuning for haswell is because it's common I think.

I was wondering because my experience is that tuning for a processor not matching the one you are actually using can sometimes greatly reduce performance.

(As a gentoo user for a decade and a half, after having recompiled the whole system many times over, I would not use anything other than ``-O2`` as general-purpose C(XX)FLAGS - anything else than that is just troubles waiting to happen)

**Comment by bluescarni** [comment]
@isuruf do you think this is good to go?

**Review by isuruf — APPROVED** [review]

**Pull request merged**
